walk-in-interview/src/main/webapp/account/create-account/applicant/script.js

```diff
@@ -51,13 +57,93 @@ function onLogOutFailure() {
 
 /** Adds all the text to the fields on this page. */
 function renderPageElements() {
-  Auth.addPhoneSignInAndSignUpUI('phone-auth', HOMEPAGE_PATH, STRINGS['new-user-info']);
+  Auth.addPhoneSignInAndSignUpUI(
+    'phone-auth', CREATE_ACCOUNT_INFO_PAGE_PATH,
+    onNewUser, onExistingUser);
 
   const backButton = document.getElementById('back');
-  backButton.innerText = COMMONG_STRINGS['back'];
+  backButton.innerText = ACCOUNT_STRINGS['back'];
+}
+
+/**
+ * The function to be executed for new user log in.
+ */
+async function onNewUser() {
+  console.log('This is a new user');
+
+  await createPreliminaryApplicantAccount();
+
+  // Informs user
+  // TODO(issue/102): replace with proper notification
+  alert(STRINGS['new-user-info']);
+
+  // Directs to the page to fill in applicant account info
+  window.location.href = CREATE_ACCOUNT_INFO_PAGE_PATH;
+}
+
+/**
+ * The function to be executed for existing user log in.
+ */
+function onExistingUser() {
+  console.log('This is not a new user');
+
+  // Informs user
+  // TODO(issue/102): replace with proper notification
+  alert(STRINGS['non-new-user-info']);
+
+  // Directs the user back to home page
+  window.location.href = HOMEPAGE_PATH;
 }
 
 const backButton = document.getElementById('back');
 backButton.addEventListener('click', (_) => {
-  window.location.href = LOGIN_HOMEPAGE_PATH;
+  window.location.href = CREATE_ACCOUNT_HOMEPAGE_PATH;
 });
+
+/**
+ * Creates a preliminary account object with phoneNumber as name for the user
+ */
+async function createPreliminaryApplicantAccount() {
+  var user = firebase.auth().currentUser;
+  if (!user) {
+    return new Promise.reject("Not signed in");
+  }
+
+  // Creates a totally empty account
+  const accountDetails = {
+    userType: USER_TYPE_APPLICANT,
+    name: user.phoneNumber,
+    // empty interest list
+  };
+
+  return fetch(API['create-applicant-account'], {
+    method: 'POST',
+    headers: {'Content-Type': 'application/json'},
+    body: JSON.stringify(accountDetails),
+  })
+      .then((response) => {
+        if (response.status == BAD_REQUEST_STATUS_CODE) {
+          setErrorMessage(/* errorMessageElementId= */'error-message',
+              /* msg= */ ACCOUNT_STRINGS['create-account-error-message'],
+              /* includesDefault= */false);
+          throw new Error(ACCOUNT_STRINGS['create-account-error-message']);
+        }
+
+        /** reset the error (there might have been an error msg from earlier) */
+        setErrorMessage(/* errorMessageElementId= */'error-message',
```

Comment:
nit: i think you really need a `clearErrorMessage('error-message');` method. it looks like you are setting an error when you are actually clearing it.

Reply:
will do in separate PR